src/zmalloc.c

```diff
@@ -102,9 +102,17 @@ static void zmalloc_default_oom(size_t size) {
 
 static void (*zmalloc_oom_handler)(size_t) = zmalloc_default_oom;
 
+#ifdef HAVE_MALLOC_SIZE
+void *extend_to_usable(void *ptr, size_t size) {
+    UNUSED(size);
+    return ptr;
+}
+#endif
+
 /* Try allocating memory, and return NULL if failed.
  * '*usable' is set to the usable size if non NULL. */
-void *ztrymalloc_usable(size_t size, size_t *usable) {
+__attribute__((malloc)) __attribute__((alloc_size(1))) 
```

Comment:
maybe that attribute should move to the public interface?
or maybe not because it'll mean we have to add it to several of them?
if we do that (move it), it'll also assist fortification without LTO.

i have a feeling we already discussed it, but wanted to raise it again.

Reply:
Do you mean to add it to `z[*]_usable`? If so `extend_to_usable()` will be useless, the compiler will only see the size passed by `z[*]_usable`.

Reply:
no, i mean to add it to all the other zmalloc functions (excluding the `_usable` one)

Reply:
@oranagra Your question led me to a bug, in any case, `ztrymalloc_usable_internal()` should not be inlined anyway, otherwise alloc_size attribute will not work.
So the attribute of ztrymalloc_usable_internal should be moved outside.

Reply:
fully CI: https://github.com/sundb/redis/actions/runs/4657525565

Reply:
ok, great.. and the fact it exposed an warning is an indication that it works better...
i suppose that if we want better fortification we can also mark them as `noinline`, but i suppose it's better to just leave it for the compiler (once https://gcc.gnu.org/bugzilla/show_bug.cgi?id=96503 will be solved).
i.e. unlike the case of `extend_to_usable` in which we must work around that bug.
so i suppose this is ready to be merged once your CI passes.
meanwhile, you can prepare the other PR with the extra CI and the compilation fixes..

Reply:
i'm not sure if noinline will bring performance impact, i need benchmark.
i'll do other thing.

Reply:
i don't think it'll bring any impact, but i suppose we shouldn't take any action here and just let the compiler do it's thing.
if the compiler is currently limited, let it be..
i assume by "other thing" you mean the CI PR... not any change to the code in this one..